src/main/java/edu/byu/cs/autograder/Grader.java

```diff
@@ -197,6 +185,39 @@ private void saveResults(TestAnalyzer.TestNode results, int numCommits) {
         submissionDao.insertSubmission(submission);
     }
 
+    /**
+     * Gets the reported score from the grader, and then adds appropriate adjustments.
+     * These adjustments include any late penalty that should be applied.
+     *
+     * @return The final score that should be saved in the Gradebook.
+     */
+    private float getFinalAdjustedScore() {
+        final int MAX_PENALIZE_DAYS_LATE = 5; // Lose credit for days late UP TO this value
```

Comment:
For the reviewer:

This code is already more readable than the existing code, but some question may be asked about why these constants are included inside this function. Another option would be to include them in a more global place where they could be easily changed or adapted in the future. I decided **not** to do this because I don't know in which global place I would put them. Additionally, I'm not certain these are the variables that will make the difference (future changes to this code may require changing these variables anyways). Therefore, I put them close to the code they adjust.

I'm happy to move them to a different location, or you can do it yourself since I gave reviewers ability to adjust the PR.

Reply:
Agreed. In several places I've added `// FIXME: ...` lines above magic numbers to remind us to go back through later and pull them out to be in a more reasonable spot, but it looks like I forgot to note this one. Thanks for pointing it out!

There are a number of configurations that should probably live in some sort of `course_settings` table. Off the top of my head this would include things like the Canvas service account API key, holidays, and this

Reply:
@pawlh See #156 which tracks the creation of a `course_settings` table.